adr/ADR-11.md

```diff
@@ -0,0 +1,55 @@
+# JetStream Encryption At Rest
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2021-07-20|
+|Author  |@derekcollison|
+|Status  |Implemented|
+|Tags    |jetstream|
+
+## Context and Problem Statement
+
+Present a way to have the nats-server itself provide encryption at rest for all stored JetStream data. This should be 
+seen as an alternative to an encrypted filesystem or block storage that would be provided by a cloud provider.
+
+
+## Design
+
+The design will allow a master key to be used to generate keys for asset encryption within JetStream. The "assets"
+are *account* information, and metadata and data for *streams* and *consumers*. Even within stream data that can be broken 
+into multiple message blocks on disk, each block will have a different asset key that will be used to encrypt.
+
+## Keys
+
+ * *EK* - External Key, provided to the nats-server configuration or via encryption service in subsequent releases.
+ * *KEK* - An encryption key derived from the *EK*. Always generated and never stored. This is generated via a PRF where 
+   `KEK := PRF(EK, IV, Context)`, e.g. `HMAC-SHA-256`, `Argon2Id`
+ * *AEK* - Asset encryption key, encrypted via a *KEK* and placed alongside the asset on disk.
+
+## Encryption
+
+Encryption will utilize an authenticated encryption AEAD. We use ChaCha20-Poly1305.
+
+## Future Support
+
+### KMS
+
+In a follow-on release we may choose to configure the nats-servers to access a KMS and not have direct access to the *EK*.
```

Comment:
For now we expect folks to use NATS configuration variables to give access to the master key to a server.